### PR TITLE
Add facade config

### PR DESCRIPTION
**Pull request by Thiblef**
https://github.com/open-orchestra/open-orchestra-base-api-bundle/pull/67
https://github.com/open-orchestra/open-orchestra-workflow-function-bundle/pull/73
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/101

### CODE REVIEW

File: ApiBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+namespace OpenOrchestra\ApiBundle\DependencyInjection;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+/**
+ * This is the class that validates and merges configuration from your app/config files
+ *
+ * To learn more see {
```
Comment (nicolasThal) [inline ApiBundle/DependencyInjection/Configuration.php]: this doc could be suppressed

File: ApiBundle/Transformer/AreaTransformer.php [hunk 2]
```diff
@@ -213,7 +214,7 @@ public function transformFromTemplate($area, TemplateInterface $template = null,
     }
 
     /**
-     * @param AreaFacade|FacadeInterface $facade
+     * @param FacadeInterface            $facade
```
Comment (nicolasThal) [inline ApiBundle/Transformer/AreaTransformer.php]: could you suppress the white spaces ?

File: UserAdminBundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace OpenOrchestra\UserAdminBundle\DependencyInjection;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+/**
+ * This is the class that validates and merges configuration from your app/config files
+ *
+ * To learn more see {
```
Comment (nicolasThal) [inline UserAdminBundle/DependencyInjection/Configuration.php]: same for the doc

**Comment by nicolasThal** [comment]
:+1:  with the passing tests and the doc updated

File: ApiBundle/DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+                    ->end()
+                    ->arrayNode('widget')
+                        ->addDefaultsIfNotSet()
+                        ->children()
+                            ->scalarNode('facade')->defaultValue('OpenOrchestra\ApiBundle\Facade\WidgetFacade')->end()
+                        ->end()
+                    ->end()
+                ->end()
+            ->end()
+        ->end();
```
Comment (itkg-ngilain) [inline ApiBundle/DependencyInjection/Configuration.php]: Maybe to get this more readable, you could use a protected method to add a facade to the tree and use it iterating an array containing the facade names and classes.

File: ApiBundle/DependencyInjection/Configuration.php [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+/**
+ * This is the class that validates and merges configuration from your app/config files
+ *
+ * To learn more see {
+ * @link http://symfony.com/doc/current/cookbook/bundles/extension.html#cookbook-bundles-extension-config-class
+ * }
+ */
+class Configuration implements ConfigurationInterface
```
Comment (itkg-alavieille) [inline ApiBundle/DependencyInjection/Configuration.php]: why an array for all type (node, node_collection, etc) and not one an arraynode (facade) and a scalarNode for each type ?

**Comment by Thiblef** [comment]
@itkg-alavieille @nicolasThal @itkg-ngilain

File: ApiBundle/Transformer/NodeGroupRoleTransformer.php [hunk 6]
```diff
@@ -31,7 +33,7 @@ public function __construct($nodeRoleGroupClass, RoleCollector $collector)
     /**
      * @param NodeGroupRoleInterface $nodeGroupRole
      *
-     * @return NodeGroupRoleFacade
```
Comment (itkg-alavieille) [inline ApiBundle/Transformer/NodeGroupRoleTransformer.php]: remove the use namespace also

**Comment by itkg-ngilain** [comment]
:+1:

File: ApiBundle/DependencyInjection/OpenOrchestraApiExtension.php [hunk 7]
```diff
@@ -19,6 +19,13 @@ class OpenOrchestraApiExtension extends Extension
      */
     public function load(array $configs, ContainerBuilder $container)
     {
+        $configuration = new Configuration();
+        $config = $this->processConfiguration($configuration, $configs);
+
+        foreach ($config['facades'] as $transformer => $facade ) {
```
Comment (itkg-alavieille) [inline ApiBundle/DependencyInjection/OpenOrchestraApiExtension.php]: psr2

File: UserAdminBundle/DependencyInjection/OpenOrchestraUserAdminExtension.php [hunk 8]
```diff
@@ -19,6 +19,13 @@ class OpenOrchestraUserAdminExtension extends Extension
      */
     public function load(array $configs, ContainerBuilder $container)
     {
+        $configuration = new Configuration();
+        $config = $this->processConfiguration($configuration, $configs);
+
+        foreach ($config['facades'] as $transformer => $facade ) {
```
Comment (itkg-alavieille) [inline UserAdminBundle/DependencyInjection/OpenOrchestraUserAdminExtension.php]: psr2

**Comment by itkg-alavieille** [comment]
:+1:  when tests run

**Pull request merged**